kong/kong.lua

```diff
@@ -252,6 +252,18 @@ function Kong.balancer()
   end
 end
 
+function Kong.rewrite()
+  core.rewrite.before()
+
+  for _, plugin in ipairs(singletons.loaded_plugins) do
+    if plugin.schema.no_consumer and plugin.schema.no_api then
+      plugin.handler:rewrite()
+    end
+  end
+
+  core.rewrite.after()
+end
+
```

Comment:
There are two ways to approach this problem

1. add the `no_api` flag: run only plugins that have `no_consumer` and `no_api` set. This causes a plugin to be configurable ONLY as a global plugin
2. execute the `rewrite` handler only for globally configured plugins, without adding the extra complexity of a `no_api` flag. This allows a plugin to be used either as a global, or api/consumer specific plugin. In the latter case the `rewrite` handler won't be used. But that is then up to the plugin author...

I find option 1 to be A) more complex due to the extra flag, 2) needlessly limiting flexibility